src/domains/wallet-app/api/payment-wallet/v1/_delete_wallet_with_pm.xml.tpl

```diff
@@ -0,0 +1,103 @@
+<policies>
+    <inbound>
+        <set-variable  name="walletToken"  value="@(context.Request.Headers.GetValueOrDefault("Authorization", "").Replace("Bearer ",""))"  />
+        <!-- Session PM START-->
+        <send-request ignore-error="true" timeout="10" response-variable-name="pm-session-body" mode="new">
+            <set-url>@($"{{pm-host}}/pp-restapi-CD/v1/users/actions/start-session?token={(string)context.Variables["walletToken"]}")</set-url>
+            <set-method>GET</set-method>
+        </send-request>
+        <choose>
+            <when condition="@(((IResponse)context.Variables["pm-session-body"]).StatusCode != 200)">
+                <return-response>
+                <set-status code="502" reason="Bad Gateway" />
+                <set-header name="Content-Type" exists-action="override">
+                    <value>application/json</value>
+                </set-header>
+                <set-body>
+                    {
+                    "title": "Error starting session",
+                    "status": 502,
+                    "detail": "There was an error starting session for input wallet token"
+                    }
+                </set-body>
+                </return-response>
+            </when>
+        </choose>
+        <set-variable name="pmSession" value="@(((IResponse)context.Variables["pm-session-body"]).Body.As<JObject>())" />
+        <!-- START delete wallet -->
+        <set-variable name="idWalletPM" value="@{
+            string walletId = context.Request.MatchedParameters["walletId"];
+            return Convert.ToInt32("0x" + walletId.Substring(walletId.Length - 12), 16);
+        }" />
+        <send-request ignore-error="false" timeout="10" response-variable-name="pmWalletDeleteResponse">
+            <set-url>@($"{{pm-host}}/pp-restapi-CD/v1/wallet/{(string)context.Variables["idWalletPM"]}")</set-url>
```

Comment:
Just to check: there is also a `DELETE /pp-restapi-CD/v1/wallet/{walletId}/{pspId}`, are we sure this works for us?

Reply:
Didn't try directly in policy, but i tried it via postman and it works. To be honest, I didn't know the difference, but at first sight I'm not so confident in this one, just for the problem to retrieve the `pspId` param. Anyway, maybe @infantesimone could help us to choice which is the API to invoke.

Reply:
the API used in PR is fine, the other API is not used (perhaps it was done for a refactoring with the introduction of paypal...)